Analysis/ingrid/private/likelihood_utils.nim

```diff
@@ -72,6 +72,61 @@ proc getInterpolatedWideDf*(df: DataFrame, num = 1000): DataFrame =
     dfLoc["Variable"] = constantColumn(tup[0][1].toStr, dfLoc.len)
     result.add dfLoc
 
+proc morphToA*(df: DataFrame, lineEnergies: seq, energy: float, offset = 1): (Tensor[int], Tensor[float]) =
+  ## generates a distribution for the appropriate energy `energy` between the
+  ## distribution below and above `energy` using linear interpolation
+  ## DF needs to have columns:
+  ## - "Hist": counts of distributions
+  ## - "Bins": bin edges of distributions
+  ## - "Dset": name of the target filter combination
+
+  let idx = max(lineEnergies.lowerBound(energy) - 1, 0)
+
+  # need idx and idx+offset
+
+  let refLow = int(lineEnergies[idx])
+  let refHigh = int(lineEnergies[idx+offset])
+ 
+
+  let refLowT = df[$refLow, float]
+  let refHighT = df[$refHigh, float]
+
+  let bins = df["bin", int]
+
+
+  var res = zeros[float](refLowT.size.int)
+  # walk over each bin and compute linear interpolation between
+  let Ediff = abs(lineEnergies[idx] - lineEnergies[idx+offset])
+  for i in 0 ..< bins.size:
+    res[i] = refLowT[i] * (1 - (abs(lineEnergies[idx] - energy)) / Ediff) +
+      refHighT[i] * (1 - (abs(lineEnergies[idx+offset] - energy)) / Ediff)
+  result = (bins, res)
```

Comment:
Later we can refactor out this type of logic from both the regular and ToA versions of this proc into a `morphImpl` that takes only the reference data, line energies and returns the bins and result. That way we only have the two procedures to do the "data extraction" from the relevant sources and the actual interpolation logic is shared. But I can do that at a later time.